app/controllers/authentication_controller.rb

```diff
@@ -1,7 +1,40 @@
 class AuthenticationController < ApplicationController
-  def sign_in; end
+  def sign_in
+    auth_request = AuthRequest.create!(
+      oauth_state: params.fetch(:state_id, SecureRandom.hex(16)),
+      oidc_nonce: SecureRandom.hex(16),
+      redirect_path: params[:redirect_path],
+    )
 
-  def callback; end
+    render json: {
+      auth_uri: OidcClient.new.auth_uri(auth_request),
+      state: auth_request.to_oauth_state,
+    }
+  end
+
+  def callback
+    auth_request = AuthRequest.from_oauth_state(params.fetch(:state))
+    head :unauthorized and return unless auth_request
+
+    client = OidcClient.new
+    tokens = client.callback(auth_request, params.fetch(:code))
+    oauth_response = client.get_ephemeral_state(
+      access_token: tokens[:access_token],
+      refresh_token: tokens[:refresh_token],
+    )
+
+    redirect_path = auth_request.redirect_path
+
+    auth_request.delete!
+
+    render json: {
+      govuk_account_session: to_account_session(oauth_response[:access_token], oauth_response[:refresh_token]),
+      redirect_path: redirect_path,
+      ga_client_id: oauth_response[:result]["_ga"],
```

Comment:
This looks good but seems far less involved than some of the things we had to do in the pervious app to preserve `_ga` around these various stages.

Is that just because we're on the same domain now?

Reply:
It's simple here because this isn't user facing, it's just passing parameters around.

In finder-frontend we still have the `redirect_with_ga` thing: https://github.com/alphagov/finder-frontend/pull/2457/files#diff-cdec550eeeb8fc63be2b5687170f594651a8d0b7f0465c9d807baef392639b6eR26